core/src/main/java/io/grpc/internal/ClientTransportFactory.java

```diff
@@ -53,6 +55,15 @@ ConnectionClientTransport newClientTransport(
    */
   ScheduledExecutorService getScheduledExecutorService();
 
+  /**
+   * Swaps to a new ChannelCredentials with all other settings unchanged. Returns null if the
+   * ChannelCredentials is not supported by the current ClientTransportFactory settings, indicating
+   * a fallback to using the default ManagedChannelRegistry.
+   */
+  @CheckReturnValue
+  @Nullable
+  ClientTransportFactory withNewChannelCredential(ChannelCredentials channelCreds);
```

Comment:
"new" here doesn't seem to provide any value; you might consider using "replacement" though. s/Credential/Credentials/.

Reply:
Done.